src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -104,6 +106,27 @@ internal DefaultVisualStudioRazorParser(RazorCodeDocument codeDocument)
         // Used in unit tests to ensure we can block background idle work.
         internal ManualResetEventSlim BlockBackgroundIdleWork { get; set; }
 
+        public override Task<RazorCodeDocument> GetLatestCodeDocumentAsync()
+        {
+            lock (GetCodeDocumentLock)
+            {
+                if (_disposed || !HasPendingChanges)
+                {
+                    // No pending changes, code document is already latest.
+                    return Task.FromResult(CodeDocument);
+                }
+
+                // In the process of parsing changes.
+
+                if (_codeDocumentTaskCompletionSource == null)
+                {
+                    _codeDocumentTaskCompletionSource = new TaskCompletionSource<RazorCodeDocument>();
```

Comment:
pass in `TaskCreationOptions.RunContinuationsAsynchronously`

Reply:
Ya tried this one out with the default config and it invoked it asynchronously so I decided not to touch it. Doesn't hurt to be explicit though.